micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -145,20 +145,17 @@ private void recordCompletionResult(ProceedingJoinPoint pjp, Counted counted, Th
     }
 
     private void record(ProceedingJoinPoint pjp, Counted counted, String exception, String result) {
-        counter(pjp, counted)
-                .tag(EXCEPTION_TAG, exception)
-                .tag(RESULT_TAG, result)
-                .tags(counted.extraTags())
-                .register(meterRegistry)
-                .increment();
-    }
-
-    private Counter.Builder counter(ProceedingJoinPoint pjp, Counted counted) {
-        Counter.Builder builder = Counter.builder(counted.value()).tags(tagsBasedOnJoinPoint.apply(pjp));
-        String description = counted.description();
-        if (!description.isEmpty()) {
-            builder.description(description);
+        try {
+            Counter.builder(counted.value())
+                    .description(counted.description().isEmpty() ? null : counted.description())
+                    .tags(counted.extraTags())
+                    .tag(EXCEPTION_TAG, exception)
+                    .tag(RESULT_TAG, result)
+                    .tags(tagsBasedOnJoinPoint.apply(pjp))
```

Comment:
Can we rely here on order of tags methods? Could tags with the same key be swapped after sort() and decouple() in Tags.of()?

Reply:
Yes the order is deterministic w/ `tags`. 
There is no place to sort or decouple tags, can you give more context for this case?

Reply:
I believe they're referring to the implementation of `Tags` which sorts and deduplicates tags with the same key, but it is designed so that later tags will take precedence over earlier tags.